mypy/util.py

```diff
@@ -467,30 +474,63 @@ class FancyFormatter:
     def __init__(self, f_out: IO[str], f_err: IO[str], show_error_codes: bool) -> None:
         self.show_error_codes = show_error_codes
         # Check if we are in a human-facing terminal on a supported platform.
-        if sys.platform not in ('linux', 'darwin'):
+        if sys.platform not in ('linux', 'darwin', 'win32'):
             self.dummy_term = True
             return
         force_color = int(os.getenv('MYPY_FORCE_COLOR', '0'))
         if not force_color and (not f_out.isatty() or not f_err.isatty()):
             self.dummy_term = True
             return
-
+        if sys.platform == 'win32':
+            self.dummy_term = self.initialize_win_colors()
+        else:
+            self.dummy_term = self.initialize_unix_colors()
+        if not self.dummy_term:
+            self.colors = {'red': self.RED, 'green': self.GREEN,
+                           'blue': self.BLUE, 'yellow': self.YELLOW,
+                           'none': ''}
+
+    def initialize_win_colors(self) -> bool:
+        # Windows ANSI escape sequences are only supported on Threshold 2 and above.
+        if sys.platform == 'win32':
```

Comment:
FWIW I would prefer the assert, instead of silently doing nothing on wrong platform.

Reply:
Unfortunately the assert does not correctly narrow the platform. I'll use both for runtime and compile time checks (and leave a note explaining why).